### PR TITLE
HIVE-25700. Prevent deletion of Notification Events post restarts.

### PR DESCRIPTION
**Pull request by ayushtkn**

### CODE REVIEW

**Comment by ayushtkn** [comment]
@pkumarsinha can you help review

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 1]
```diff
@@ -752,6 +752,9 @@ public static ConfVars getMetaConf(String name) {
     EVENT_DB_LISTENER_CLEAN_INTERVAL("metastore.event.db.listener.clean.interval",
             "hive.metastore.event.db.listener.clean.interval", 7200, TimeUnit.SECONDS,
             "sleep interval between each run for cleanup of events from the database listener queue"),
+    EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL("metastore.event.db.listener.clean.startup.wait.interval",
+        "hive.metastore.event.db.listener.clean.startup.wait.interval", 1, TimeUnit.DAYS,
+        "wait interval post start of Cleaner thread after which the cleaner thread starts to work"),
```
Comment (pkumarsinha) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:757]: a) Capitalize "wait".
b) Is there a difference between Cleaner thread start time vs HMS process uptime(not considering time taken to launch a thread which idea would be very minimal)? If not, then referring to process start time would be better to convey the same.

File: hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java [hunk 2]
```diff
@@ -1417,10 +1426,27 @@ private void process(NotificationEvent event, ListenerEvent listenerEvent) throw
...
+      LOG.info("Wait interval is {}", waitInterval);
+      if (waitInterval > 0) {
+        try {
+          LOG.info("Cleaner Thread Restarted and {} or {} is configured. So cleaner thread will startup post waiting "
+                  + "{} ms", EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL,
+              EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL.getHiveName(), waitInterval);
+          Thread.sleep(waitInterval);
+        } catch (InterruptedException e) {
+          LOG.error("Failed during the initial wait before start.", e);
+          Thread.currentThread().interrupt();
```
[the remark this answers is not in the document]
Comment (ayushtkn) [reply hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java]: As discussed, Have wrapped this isTest check

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java (hunk 1 above)
Comment (ayushtkn) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java]: Done

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java [hunk 3]
```diff
@@ -252,6 +254,8 @@ static void internalBeforeClassSetup(Map<String, String> additionalProperties)
     metaStoreClientMirror = new HiveMetaStoreClient(hconfMirror);
 
     PersistenceManagerProvider.setTwoMetastoreTesting(true);
+    MetastoreConf.setTimeVar(hconf, EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL, 0, TimeUnit.SECONDS);
+    MetastoreConf.setTimeVar(hconfMirror, EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL, 0, TimeUnit.SECONDS);
```
Comment (pkumarsinha) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java:258]: Are these lines needed. Isn't the config optional anyway or the idea is to test >0 case in default case?

**Review by pkumarsinha — COMMENTED** [review]
Looks good otherwise, just one minor comment

**Review by pkumarsinha — APPROVED** [review]
+1 LGTM

**Pull request merged**